packages/@lightningjs/ui-components/src/components/Badge/Badge.js

```diff
@@ -68,29 +70,53 @@ export default class Badge extends Base {
     this._updateText();
     this._updateIcon();
     this._updateLayout();
+    this._updateVisibility();
+  }
+
+  _updateVisibility() {
+    if (!this.title && !this.icon) {
+      this.alpha = 0.001;
```

Comment:
I think this could even be alpha 0. If neither exist, we should be OK to totally de-render it